### PR TITLE
Add config options pool_sup_intensity and pool_sup_period

### LINKED ISSUE #64: one_for_all strategy with the worker pool supervisor
This is a bit long, because I'm not sure what solution I'd like. I have the following situation:

There is a single producer (reads an AWS Kinesis stream) feeding a set of consumers (a worker_pool that writes the data to the DB and to S3). 

The producer produces data, and also checkpoint requests. The consumers consume the data, and when they've fully consumed everything up to the checkpoint request, they notify the producer. The producer needs to see responses from all workers before it can checkpoint.

My problem is that if a single worker dies I want it to restart the producer, doing `one_for_all` across the pool of workers and the producer.

When I create the worker_pool, I request a restart intensity of `0` so that any worker failure restarts the pool supervisor. However, this doesn't help because there's a supervisor above that which I can't configure: https://github.com/inaka/worker_pool/blob/master/src/wpool_pool.erl#L345

Here are a few ideas for things that would fix this for me
1. Make the `wpool_pool` supervisor configurable, as well as the `wpool_process_sup` supervisor. This would be easy, but seems confusing and ugly to have two config options for supervisors.
2. Have the `wpool_pool` supervisor restart at 0 intensity. This could be annoying if `wpool_time_checker` or `wpool_queue_manager` were crashing.
3. Export a function that makes the childspec for `wpool_pool`, so I can attach it to my supervisor directly. Same problem as 2, though the crash output would look a little different.

Any other thoughts? I'm happy to write the code if you've got a solution that you like. Or am I just misusing worker_pool or OTP?

### PR DESCRIPTION
**Pull request by waisbrot**
Options for altering the period and intensity of the `wpool_pool`
supervisor.

Changed the README to include these two options and the `strategy`
option that was already available.

Fixes #64 

Comments on the option names or the README doc, @elbrujohalcon ?

### CODE REVIEW

**Comment by elbrujohalcon** [comment]
I like what you did, @waisbrot .
Any chance you can add a test case for this?

**Comment by waisbrot** [comment]
Sure, I'll try adding some tests.

**Comment by waisbrot** [comment]
I've added a pair of tests for this that are working for me. I don't know Common Test all that well, so I'm not sure how to properly clean up the pools I make.

I can squish these commits down once you like them (unless you like 'em unsquished).

File: test/wpool_process_SUITE.erl [hunk 1]
```diff
@@ -101,3 +105,34 @@ call(_Config) ->
...
+    timer:sleep(500),
+
+    ct:log("Check that the pool is working"),
+    true = erlang:is_process_alive(Pid),
+    hello = wpool:call(Pool, hello).
+
+-spec pool_norestart_crash(config()) -> _.
+pool_norestart_crash(_Config) ->
+    Pool = pool_norestart_crash,
+    {ok, Pid} = wpool:start_pool(Pool, [{workers, 2}, {worker, {crashy_server, []}}, {strategy, {one_for_all, 0, 10}}, {pool_sup_intensity, 0}, {pool_sup_period, 10}]),
```
Comment (elbrujohalcon) [inline test/wpool_process_SUITE.erl]: I'm not sure why @gadgetci is not warning us about this line, but when I run `$ rebar3 ct` or `$ elvis rock` on my computer, I do get warnings about it.
Can you reduce it to 80 characters, please?

File: test/wpool_process_SUITE.erl [hunk 2]
```diff
@@ -101,3 +105,34 @@ call(_Config) ->
   ok3 = wpool_process:call(Pid, {stop, normal, ok3, state}, 5000),
   timer:sleep(1000),
   false = erlang:is_process_alive(Pid).
+
+-spec pool_restart_crash(config()) -> _.
+pool_restart_crash(_Config) ->
+    Pool = pool_restart_crash,
+    {ok, Pid} = wpool:start_pool(Pool, [{workers, 2}, {worker, {crashy_server, []}}]),
```
Comment (elbrujohalcon) [inline test/wpool_process_SUITE.erl]: I'm not sure why @gadgetci is not warning us about this line, but when I run `$ rebar3 ct` or `$ elvis rock` on my computer, I do get warnings about it.
Can you reduce it to 80 characters, please?

**Comment by waisbrot** [comment]
I've broken up the long lines.

**Comment by elbrujohalcon** [comment]
Thanks, @waisbrot :)

**Pull request merged**